python/ray/train/torch/torch_trainer.py

```diff
@@ -103,65 +106,106 @@ def train_loop_per_worker():
                         inputs, labels = torch.unsqueeze(batches["x"], 1), batches["y"]
                         output = model(inputs)
-                        loss = loss_fn(output, labels)
+
+                        # Get outputs as the same dimension as labels
+                        loss = loss_fn(output.squeeze(), labels)
+
+                        # Zero out grads, do backward, and update optimizer
                         optimizer.zero_grad()
                         loss.backward()
                         optimizer.step()
-                        print(f"epoch: {epoch}, loss: {loss.item()}")
 
-                    session.report(
-                        {},
-                        checkpoint=Checkpoint.from_dict(
-                            dict(epoch=epoch, model=model.state_dict()
-                        ),
+                        # Print what's happening with loss per 30 epochs
+                        if epoch % 30 == 0:
+                            print(f"epoch: {epoch}/{num_epochs}, loss: {loss:.3f}")
+
+                    # Report and record metrics, checkpoint model at end of each
+                    # epoch
+                    session.report({"loss": loss.item(), "epoch": epoch},
+                                         checkpoint=Checkpoint.from_dict(
+                                         dict(epoch=epoch, model=model.state_dict()))
                     )
 
+            torch.manual_seed(42)
             train_dataset = ray.data.from_items(
-                [{"x": x, "y": 2 * x + 1} for x in range(200)]
+                     [{"x": x, "y": 2 * x + 1} for x in range(2000)]
             )
-            scaling_config = ScalingConfig(num_workers=3)
+
+            # Define scaling and run configs
             # If using GPUs, use the below scaling config instead.
             # scaling_config = ScalingConfig(num_workers=3, use_gpu=True)
+            scaling_config = ScalingConfig(num_workers=num_workers)
+            run_config = RunConfig(checkpoint_config=CheckpointConfig(num_to_keep=1))
+
             trainer = TorchTrainer(
                 train_loop_per_worker=train_loop_per_worker,
                 scaling_config=scaling_config,
+                run_config=run_config,
                 datasets={"train": train_dataset})
+
             result = trainer.fit()
 
+            # Get the loss metric from TorchCheckpoint tuple data dictionary
+            best_checkpoint_loss = result.metrics['loss']
+            # print(f"best loss: {best_checkpoint_loss:.4f}")
+
+
+            # Assert loss is less 0.09
+            assert best_checkpoint_loss <= 0.09
+
+    .. testoutput::
+
+        ...
+
     Args:
+
         train_loop_per_worker: The training function to execute.
             This can either take in no arguments or a ``config`` dict.
-        train_loop_config: Configurations to pass into
+            train_loop_config: Configurations to pass into
```

Comment:
The indentation should be kept here, right? Otherwise it won't render correctly :)

Reply:
The Args name: should be indented on the same level. That is:

```
Args:
        arg_1:  ...
        arg_2: ...
```

Reply:
<img width="774" alt="Screen Shot 2022-11-21 at 3 25 39 PM" src="https://user-images.githubusercontent.com/1117597/203178161-2f7ef76a-5456-4a2b-86fa-9cc58359fb66.png">
They seem to render properlyl